mmdet3d/models/middle_encoders/voxel_set_abstraction.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import ConvModule
+from mmcv.ops.furthest_point_sample import furthest_point_sample
+from mmengine.model import BaseModule
+
+from mmdet3d.registry import MODELS
+from mmdet3d.utils import InstanceList
+
+
+def bilinear_interpolate_torch(inputs, x, y):
```

Comment:
add type-hint here

Reply:
Can we use `torch.grid_sample` to get the feature for certain (x,y) position from Bev feature?

Reply:
Yeah, I don't make sure whether `torch.grid_sample` is available, because when I used it to test the model inference result I find it was different from the source code, so I modified it, and I will check whether it influenced the model training.